### PR TITLE
Misc small improvements and simplifications

### PR DESCRIPTION
**Pull request by cobratbq**
Misc small improvements.

- Fix javadoc linter warning.
- Consistent coding style.
- Use SLF4J logger and remove constructor injection of Log instance.
- Unnecessary nesting.

Let me know if you want it split up, or you're not interested or whatever. This is just what I gathered while investigating the issue regarding stricter treatment of missing keysmap location.

Also, let me know if you want this squashed too. I did not do it for now, as it makes clearer what sort of changes were made.

### CODE REVIEW

File: src/main/java/org/simplify4u/plugins/ArtifactResolver.java [hunk 1]
```diff
@@ -358,11 +358,11 @@ private Artifact resolveSignature(Artifact artifact, SignatureRequirement requir
...
         if (!resolution.isSuccess()) {
             if (resolution.hasMissingArtifacts()) {
-                log.warn("Missing artifacts for " + artifact.getId() + ": " + resolution.getMissingArtifacts());
+                LOG.warn("Missing artifacts for {}: {}", artifact.getId(), resolution.getMissingArtifacts());
             }
             resolution.getExceptions().forEach(e -> {
-                log.warn("Failed to resolve transitive dependencies for " + artifact.getId() + ": " + e.getMessage());
-                log.debug(e);
+                LOG.warn("Failed to resolve transitive dependencies for {}: {}", artifact.getId(), e.getMessage());
+                LOG.debug("Exceptions in artifact resolution.", e);
```
Comment (slawekjaranowski) [inline src/main/java/org/simplify4u/plugins/ArtifactResolver.java:365]: Remove duplicate logging in debug level

File: src/main/java/org/simplify4u/plugins/ArtifactResolver.java [hunk 2]
```diff
@@ -393,8 +393,8 @@ private Artifact resolveArtifact(Artifact artifact) {
         final ArtifactResolutionResult result = request(artifact, remoteRepositories);
         if (!result.isSuccess()) {
             result.getExceptions().forEach(e -> {
-                log.warn("Failed to resolve " + artifact.getId() + ": " + e.getMessage());
-                log.debug(e);
+                LOG.warn("Failed to resolve {}: {}", artifact.getId(), e.getMessage());
+                LOG.debug("Exceptions in artifact resolution.", e);
```
Comment (slawekjaranowski) [inline src/main/java/org/simplify4u/plugins/ArtifactResolver.java:397]: Remove duplicate logging in debug level

File: src/main/java/org/simplify4u/plugins/ArtifactResolver.java (hunk 1 above)
Comment (cobratbq) [reply src/main/java/org/simplify4u/plugins/ArtifactResolver.java:365]: There is no method available that only accepts a throwable to log just the exception. That's why I added the line. If we remove the line in total, we do not have any exception info at all. Any ideas?
I mean, we can always remove the debug-logging in total. In that case we have the error _message_ but not the complete stacktrace.

Comment (slawekjaranowski) [reply src/main/java/org/simplify4u/plugins/ArtifactResolver.java:365]: I thought, that we can remove log with debug and exception stacktrace at all.
I think that stacktrace in this point is not really helpful.

**Pull request merged**